docs/chapters/04-1.md

```diff
@@ -0,0 +1,218 @@
+
+Each layer has a matrix containing just one kernel that replicates multiple times. This kind of matrix is called toeplitz matrix. In every toeplitz matrix, each descending diagonal from left to right is constant. As for the toeplitz matrices that we use here, they are also sparse matrices.
+
+Given the first kernel $\mathbf{a}^{(1)}$ and the input vector $\mathbf{x}$, the first entry in the output given by this layer is $a_1^{(1)} x_1 + a_2^{(1)} x_2 + a_3^{(1)}x_3$. Therefore, the whole output vector is like
+$$
+\begin{bmatrix}
+\mathbf{a}^{(1)}x[1:3]\\
+\mathbf{a}^{(1)}x[2:4]\\
+\mathbf{a}^{(1)}x[3:5]\\
+\vdots
+\end{bmatrix}
+$$
+For other kernels (e.g. $\mathbf{a}^{(2)}$ and $\mathbf{a}^{(3)}$) in following convolutional layers, we can apply the same matrix multiplication method and get the similar results.
+
+
+## [Listening to Convolutions - Jupyter Notebook](0:33:01-0:59:59)
+
+The Jupyter Notebook can be found [here](https://github.com/Atcold/pytorch-Deep-Learning/blob/master/07-listening_to_kernels.ipynb). Note the names of github repository and conda environment have changed.
+
+In this notebook, we are going to explore Convolution as 'running scalar product'.
+
+Importing library librosa and we can load data $\mathbf{x}$ and sampling rate. In this case, there are 70641 samples, learning rate is 22.05kHz and total time is 3.2s. The imported audio signal is wavy (refer to Fig1) and we can guess what it sounds like from the amplitude of y axis. Listening to x(t), the audio signal is actually the sound played when turning off the Windows system (refer to Fig2). 
+![Fig1 Audio signal](https://i.imgur.com/LLdvx8d.png)
+![Fig2 Music notes](https://i.imgur.com/suWrplO.png)
+
+To get notes from wave from, Fourier transform (FT) is a good guess, however, if FT is played on the whole signal, notes would come out together, the is hard to figure out the exact time and location of each pitch. Localized FT is needed, which is called spectroscope. As is shown in the spectrogram (refer to Fig3), different pitched peak at different values (e.g. first pitch peaks at 1600). Concatenate the four pitches based on their frequencies, we get the reconstruction. 
+![Fig3 Audio sugnal and its spectrum](https://i.imgur.com/i5oWuZH.png)
+
+Convolution of signal with different pitches can help to extract all notes from a given piece, auido matches specific kernels. The spectrograms of origin signal and the conctatenation (refer to Fig4) is shown, as well as the frequencies of real signal and four pitches (refer to Fig5). The plot and sound of convolution of four kernels indicate convolution performance is pretty good (refer to Fig6). 
+
+![Fig4 Spectrogram of real signal and concatenation](https://i.imgur.com/YTyVcH7.png)
+![Fig5 First melody's note](https://i.imgur.com/O4narOi.png)
+![Fig6 COnvolution of four kernels](https://i.imgur.com/R6iJh9N.png)
```

Comment:
All images should be centered and have a caption and be loaded from the repo, not from online, somewhere.